### PR TITLE
Add devise For user PR

### PR DESCRIPTION
**Pull request by YoutLone**
### In this PR contain the following : 

- Implement the `ApplicationController`
- Install and setup the devise
- New user can register.
- User logs in with a combination of email and password.
- Hashed password will be stored in the database.
- Ask for confirmation of email.
- Can reset password.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
```
Comment (iambenkis) [inline app/models/user.rb:5]: - [ ] Kudos for importing the required modules for the devise 👍 👍 👍

File: app/views/pages/_header.html.erb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<nav class="navbar navbar-expand-lg bg-success border-bottom border-body" data-bs-theme="dark">
+  <div class="container-fluid">
+    <%= link_to "BlogApp", root_path, class:"navbar-brand"%>
+  </div>
+</nav>
```
Comment (iambenkis) [inline app/views/pages/_header.html.erb:5]: - [ ] [OPTIONAL] Good job adding the devise authentication in your blog app, everything about the Registration and the Login works properly. At the same time , I was able to detect an issue when I wanted to create a new post ,your bug is supposed to be working , 👍 Kindly fix that issue to allow users to create posts 😄 

Capture1

Capture

**Review by iambenkis — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- Require module imported :heavy_check_mark:
- Registration and Login implemented :heavy_check_mark: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @iambenkis in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/pages/_header.html.erb (hunk 2 above)
Comment (YoutLone) [reply app/views/pages/_header.html.erb:5]: Thank you @iambenkis for your review and I will fix it.

**Pull request merged**